### PR TITLE
bug 1959513: add default policy, use lib go audit

### PR DESCRIPTION
**Pull request by EmilyM1**
This is the removal of the audit pkg internal for the cluster-kube-apiserver operator so it can use library-go
Then it bumps the version of library-go that has the changes needed so library-go can be used which are here 
https://github.com/openshift/library-go/pull/1025/

### CODE REVIEW

File: pkg/cmd/render/render.go [hunk 1]
```diff
@@ -20,10 +20,10 @@ import (
 	"k8s.io/klog/v2"
 
 	kubecontrolplanev1 "github.com/openshift/api/kubecontrolplane/v1"
-	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/audit"
 	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/v410_00_assets"
 	genericrender "github.com/openshift/library-go/pkg/operator/render"
 	genericrenderoptions "github.com/openshift/library-go/pkg/operator/render/options"
+	libgoassets "github.com/openshift/library-go/pkg/operator/apiserver/audit"
```
Comment (sttts) [inline pkg/cmd/render/render.go]: why assets?

File: pkg/cmd/render/render_test.go [hunk 2]
```diff
@@ -21,8 +21,8 @@ import (
 	configv1 "github.com/openshift/api/config/v1"
 	kubecontrolplanev1 "github.com/openshift/api/kubecontrolplane/v1"
 	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/configobservation/configobservercontroller"
-	genericrenderoptions "github.com/openshift/library-go/pkg/operator/render/options"
 	libgoassets "github.com/openshift/library-go/pkg/operator/apiserver/audit"
+	genericrenderoptions "github.com/openshift/library-go/pkg/operator/render/options"
```
Comment (sttts) [inline pkg/cmd/render/render_test.go:25]: please squash this commit

File: vendor/modules.txt [hunk 3]
```diff
@@ -118,8 +118,6 @@ github.com/imdario/mergo
 github.com/inconshreveable/mousetrap
 # github.com/json-iterator/go v1.1.10
 github.com/json-iterator/go
-# github.com/konsorten/go-windows-terminal-sequences v1.0.3
-## explicit
```
Comment (sttts) [inline vendor/modules.txt]: squash into bump commit

**Comment by sttts** [comment]
Looks good overall.

/assign @tkashem

File: pkg/cmd/render/render_test.go (hunk 2 above)
Comment (EmilyM1) [reply pkg/cmd/render/render_test.go:25]: good point. Done

**Comment by tkashem** [comment]
@EmilyM1 did you run `make update-bindata` to remove the old audit asset from the operator asset package? I don't see the removed bits here

**Comment by tkashem** [comment]
also, we probably need to add these bits:
https://github.com/openshift/cluster-kube-apiserver-operator/pull/1090/commits/2fc08d73d12bb4ffb83c8fa6e16e0e72b17198d0#diff-0d623dfd885adb20f991bda4c2453aebd732ca6dbb4d1d4be6e79805c3b48de6R115-R124

https://github.com/openshift/cluster-kube-apiserver-operator/pull/1090/commits/2fc08d73d12bb4ffb83c8fa6e16e0e72b17198d0#diff-0d623dfd885adb20f991bda4c2453aebd732ca6dbb4d1d4be6e79805c3b48de6R142-R143

File: pkg/cmd/render/render.go [hunk 4]
```diff
@@ -20,8 +20,8 @@ import (
 	"k8s.io/klog/v2"
 
 	kubecontrolplanev1 "github.com/openshift/api/kubecontrolplane/v1"
-	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/audit"
 	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/v410_00_assets"
+	libgoassets "github.com/openshift/library-go/pkg/operator/apiserver/audit"
```
Comment (sttts) [inline pkg/cmd/render/render.go:24]: libgoaudit would be better.

File: pkg/operator/starter.go [hunk 5]
```diff
@@ -131,7 +140,7 @@ func RunOperator(ctx context.Context, controllerContext *controllercmd.Controlle
 			"v4.1.0/kube-apiserver/localhost-recovery-client-crb.yaml",
 			"v4.1.0/kube-apiserver/localhost-recovery-sa.yaml",
 			"v4.1.0/kube-apiserver/localhost-recovery-token.yaml",
-			"v4.1.0/kube-apiserver/audit-policies-cm.yaml",
+			"pkg/operator/apiserver/audit/manifests/audit-policies-cm.yaml",
```
Comment (sttts) [inline pkg/operator/starter.go:143]: how do we do this in other operators? Seeing audit here is a little strange.

Comment (EmilyM1) [reply pkg/operator/starter.go:143]: https://github.com/openshift/cluster-openshift-apiserver-operator/blob/master/pkg/operator/starter.go#L193

File: pkg/operator/starter.go [hunk 6]
```diff
@@ -142,11 +151,11 @@ func RunOperator(ctx context.Context, controllerContext *controllercmd.Controlle
 			"v4.1.0/kube-apiserver/localhost-recovery-client-crb.yaml",
 			"v4.1.0/kube-apiserver/localhost-recovery-sa.yaml",
 			"v4.1.0/kube-apiserver/localhost-recovery-token.yaml",
-			"v4.1.0/kube-apiserver/audit-policies-cm.yaml",
 			"v4.1.0/kube-apiserver/apiserver.openshift.io_apirequestcount.yaml",
 			"v4.1.0/alerts/api-usage.yaml",
 			"v4.1.0/alerts/cpu-utilization.yaml",
 			"v4.1.0/alerts/kube-apiserver-requests.yaml",
+			"v4.1.0/kube-apiserver/audit-policies-cm.yaml",
```
Comment (stlaz) [inline pkg/operator/starter.go:158]: so this is _not_ going to use the library-go's audit-policies?

Comment (EmilyM1) [reply pkg/operator/starter.go:158]: Was troubleshooting to see how much CI fail is based on audit.cm location

File: pkg/operator/starter.go [hunk 7]
```diff
@@ -27,6 +26,8 @@ import (
 	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/terminationobserver"
 	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/v410_00_assets"
 	"github.com/openshift/library-go/pkg/controller/controllercmd"
+	libgoaudit "github.com/openshift/library-go/pkg/operator/apiserver/audit"
+	libgoauditasset "github.com/openshift/library-go/pkg/operator/apiserver/audit/bindata"
```
Comment (tkashem) [inline pkg/operator/starter.go:30]: `libgoauditasset` should be unused

File: pkg/operator/starter.go (hunk 5 above)
Comment (tkashem) [reply pkg/operator/starter.go:149]: @EmilyM1 try `libgoaudit.AuditPoliciesConfigMapFileName` here

File: pkg/operator/starter.go [hunk 8]
```diff
@@ -142,11 +142,11 @@ func RunOperator(ctx context.Context, controllerContext *controllercmd.Controlle
 			"v4.1.0/kube-apiserver/localhost-recovery-client-crb.yaml",
 			"v4.1.0/kube-apiserver/localhost-recovery-sa.yaml",
 			"v4.1.0/kube-apiserver/localhost-recovery-token.yaml",
-			"v4.1.0/kube-apiserver/audit-policies-cm.yaml",
 			"v4.1.0/kube-apiserver/apiserver.openshift.io_apirequestcount.yaml",
 			"v4.1.0/alerts/api-usage.yaml",
 			"v4.1.0/alerts/cpu-utilization.yaml",
 			"v4.1.0/alerts/kube-apiserver-requests.yaml",
+			"libgoaudit.AuditPoliciesConfigMapFileName",
```
Comment (tkashem) [inline pkg/operator/starter.go:149]: not a literal, it should not be quoted, plain `libgoaudit.AuditPoliciesConfigMapFileName`

**Pull request merged**